code/game/turfs/turf.dm

```diff
@@ -186,11 +187,14 @@
 			lighting_build_overlays()
 		else
 			lighting_clear_overlays()
-	
+
 	obscured = old_obscured
 
 	return W
 
+/turf/proc/PreChange()
+	return
+
```

Comment:
Shouldn't this be called `BeforeChange`, to go alongside `AfterChange`?

Also, there's probably some `ChangeTurf` overrides that need to become one or the other, such as the one on walls that deletes wallrot.